### DISCUSSION TITLE
NAME OF SCP training layer are all identical

### DISCUSSION THREAD

**Discussion by GITHUBAntoineDENIS** [semiautomaticgit/SemiAutomaticClassificationPlugin#357] [General]
Hello,
When creating "SCP training layer" for calibration or validation ROI drawing, all SCP training layers created show the same name in the "layers" pannel of qgis and do not use the name given when creating the roi file in the working directory.

So if we create first calibration roi for classification, ans then validation roi, they will have both the same name, and this will make it difficult to know which file to choose when we have to indicate the validation roi to use in the accuracy assessment window > select reference vector.

--> it would be easier if the "scp training layer" display the name given rather than "SCP training layer".
Best regards,
Antoine

**Comment by semiautomaticgit** [comment]
Hello @GITHUBAntoineDENIS ,

thank you for your suggestion.
However, it is actually designed to display the same name because it is a layer managed by the SCP and should not be edited directly (it is only used to display existing ROIs). In the future I'll try to hide the layer from the layer list.

**Comment by GITHUBAntoineDENIS** [comment]
Hello. OK. But this makes it difficult to know which file to choose when we have to indicate the validation roi file to use in the accuracy assessment window > select reference vector. We have 2 files with the same name  "scp training layer" (roi calibration and roi validation), and so how to know which one to choose? What I did is to rename one in the qgis layers pannel. Another way would be to delete the calibration roi from layers pannel before going on with the validation. 
Just to bring you detailed information on the encountered problem.
Best regards,
Antoine

**Comment by semiautomaticgit** [comment]
Hello @GITHUBAntoineDENIS ,
thank you for your feedback.

> Hello. OK. But this makes it difficult to know which file to choose when we have to indicate the validation roi file to use in the accuracy assessment window > select reference vector. We have 2 files with the same name "scp training layer" (roi calibration and roi validation), and so how to know which one to choose? 

The "scp training layer" is the one loaded in the SCP dock, so there should be only one at a time in the layers. If you replace the training input in SCP dock, the "scp training layer" is automatically replaced, so there should be no confusion.

>What I did is to rename one in the qgis layers pannel.
I suggest to avoid renaming the layer, as it could cause some malfunctions in SCP.
I would recommend to create a different geopackage for validation if it cause confusion.

**Comment by GITHUBAntoineDENIS** [comment]
Hello Lucas,
Thanks for the reply. Your explanations above did not "fit" with what I saw in QGIS, so I investigated a little bit to understand why this difference. 
In fact, when doing the whole process non stop in QGIS (calibration, then validation) without closing/reopening the project, the "scp training layer" behave like you say: there is only one file in the layers pannel, and when opening a new roi, the training layer is automatically updated.
But if i create a "scp training layer" for calibration, then closing qgis, then reopening the project, at the reopening, a new "scp training layer" is automatically added in the layer pannel. It can be twice the same roi file if only one file has be created so far. But we also might finally end with two "scp training layer" corresponding to one calibration file and one validation file if we are working with 2 roi files. so we get twice the same name for 2 different files. you can see that in the image below. So the "problem" happens when closing/reopening  a qgis project. 
Best regards,
Antoine 
![image](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/assets/44610746/b17c23a6-ff86-4de0-ab9a-10da8f3a3ef8)

![image](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/assets/44610746/41064624-3059-4128-9f66-c47b494ba950)
